perun-auditlogger/src/main/java/cz/metacentrum/perun/auditlogger/logger/impl/EventLoggerImpl.java

```diff
@@ -0,0 +1,197 @@
+package cz.metacentrum.perun.auditlogger.logger.impl;
+
+import cz.metacentrum.perun.audit.events.AuditEvent;
+import cz.metacentrum.perun.auditlogger.logger.EventLogger;
+import cz.metacentrum.perun.auditlogger.service.AuditLoggerManager;
+import cz.metacentrum.perun.auditparser.AuditParser;
+import cz.metacentrum.perun.core.api.AuditMessage;
+import cz.metacentrum.perun.core.api.Perun;
+import cz.metacentrum.perun.core.api.PerunSession;
+import cz.metacentrum.perun.core.api.exceptions.InternalErrorException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+@org.springframework.stereotype.Service(value = "eventLogger")
+public class EventLoggerImpl implements EventLogger, Runnable {
+
+	private final static Logger log = LoggerFactory.getLogger(EventLoggerImpl.class);
+
+	private static final DateFormat DATE_FORMAT = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
+
+	private static final String SYSLOG_LOGGER_NAME = "syslog-logger";
+	
+	private static final Logger syslog = LoggerFactory.getLogger(SYSLOG_LOGGER_NAME);
+	
+	private static final Map<Class<?>,Class<?>> mixinMap = new HashMap<>();
+	private static final ObjectMapper mapper = new ObjectMapper();
+
+	static {
+		mapper.enableDefaultTyping();
+		// TODO - skip any problematic properties using interfaces for mixins
+		mapper.setMixIns(mixinMap);
+	}
+
+	@Autowired
+	private AuditLoggerManager auditLoggerManager;
+
+	private PerunSession perunSession;
+	private Perun perun;
+	private int lastProcessedIdNumber;
+
+	private boolean running = false;
+
+	@Override
+	public void run() {
+
+
+		running = true;
+		AuditMessage message = null;
+		List<AuditMessage> messages;
+
+		try {
+			perunSession = auditLoggerManager.getPerunSession();
+			perun = auditLoggerManager.getPerunBl();
+			
+			if (lastProcessedIdNumber == 0) {
+				loadLastProcessedId();
+			}
+
+			messages = null;
+
+			//If running is true, then this process will be continuously
+			while (running) {
+
+				int sleepTime = 1000;
+				//Waiting for new messages. If consumer failed in some internal case, waiting until it will be repaired (waiting time is increases by each attempt)
+				while(messages == null) {
+					try {
+						//IMPORTANT STEP1: Get new bulk of messages
+						messages = perun.getAuditMessagesManager().pollConsumerMessages(perunSession, auditLoggerManager.getConsumerName(), lastProcessedIdNumber);
```

Comment:
I think we can use BL instead of Entry layer to avoid authorization check, which might be the reason you used ``perunLdapc`` principal in the session, which is by default configured to be perun admin.

Reply:
Ok. Which makes the authorization checks rather superficial imho.

Reply:
Well, I don't think, that internal component (calling single method) should be perun admin just because the called method requires it. We could probably create new role, but it should be then true also for other internal principals, etc. Also entry generally creates new transaction, which might not be desired in other components (overhead). So we probably should improve all this logic in the future, but it is outside of the scope of this pull-request.